### ISSUE TITLE
clangd support?

### ISSUE THREAD

**Issue by Xeverous** [Cevelop/Issues#131]
[clangd](https://clangd.llvm.org) offloads a lot of IDE work while at the same time being implemented as a part of compiler infrastructure. The [LSP](https://microsoft.github.io/language-server-protocol/) has already many predefined calls for code navigation and some editing.

Some IDEs (most notably VS Code and the ones by Jet Brains) have moved towards using LSP instead of writing their own code parsers.

I suggest that Cevelop could also take this opportunity. I assume that refactoring features are relying on the built-in parser but:

- the integration of clangd is relatively small compared to writing analogical tooling in the IDE itself (you just have to exchange JSONs, the real work is already done in clangd)
- clangd (in order to understand code properly) requires full compiler information (`compile_commands.json`) but CMake can deliver it for any CMake-based project with 1 command line option

In short, I think Cevelop can gain a lot of value just by reading CMake-generated compiler commands and communicating with clangd.
